guidelines/coding-guidelines.md

```diff
@@ -11,3 +11,59 @@ For public facing interfaces we favor read-only collections and enumerables. For
 * Avoid using `System.Linq`
 * Avoid using `foreach` over collections that do not have a struct enumerator
 
+## Dependencies
+
+### 3rd party dependencies
+
+#### Package dependencies
+
+We want to compile against the lowest major to catch breaking changes
+
+PackageReference: `[Major.0.0, Major+1)`
+
+#### Tests
+
+We want to compile against the latest version to catch breaking changes and bugs. This is the likely version that our users would be on.
+
+PackageReference: `[Major.*, Major+1)`
```

Comment:
Is there a risk of the tests taking advantage of new behaviours, while the package remains dependent on older versions which display different behaviours? Currently in RabbitMQ we test against the same version that that the package is built against.

Reply:
I don't see how that solves anything? Until we have tests that can be rerun against the entire set of supported core dependency versions we can't be sure that they work?

Reply:
@andreasohlund yep, this is a continuous head-scratcher. Disregarding execution time and resources, we'd run our test on all possible permutations. On reflection, it is probably best to run the tests against latest, rather than earliest, compatible core. It's the lesser of the two evils.

However, I misspoke in my original comment. At some point, in RabbitMQ, we also started to run our tests against the latest version of Core, so please diregard this thread.